assets/src/stories-editor/components/template-inserter/template-preview.js

```diff
@@ -8,47 +8,37 @@ import PropTypes from 'prop-types';
  */
 import { withSafeTimeout } from '@wordpress/compose';
 import { Spinner } from '@wordpress/components';
-import { Component } from '@wordpress/element';
+import { useState, useEffect } from '@wordpress/element';
 
 /**
  * Internal dependencies
  */
 import { BlockPreview } from '../';
 
-class TemplatePreview extends Component {
-	constructor( ...args ) {
-		super( ...args );
+const TemplatePreview = ( { item } ) => {
+	const [ shouldLoad, setShouldLoad ] = useState( false );
 
-		this.state = {
-			shouldLoad: false,
-		};
-	}
-
-	componentDidMount() {
-		if ( ! this.state.shouldLoad ) {
+	useEffect( () => {
+		if ( ! shouldLoad ) {
 			// @todo Look into React Concurrent mode to replace this once it gets available.
 			// Set timeout to cause a small latency between loading the templates, otherwise they all try to load instantly and cause a lag.
 			this.props.setTimeout( () => {
-				this.setState( { shouldLoad: true } );
+				setShouldLoad( true );
 			}, 100 );
 		}
-	}
-
-	render() {
-		if ( ! this.state.shouldLoad ) {
-			return <Spinner />;
-		}
-
-		const { item } = this.props;
+	}, [ shouldLoad ] );
```

Comment:
This whole effect can be replaced with:

```
	useEffect( () => setTimeout( setShouldLoad, 100, true ), [] );
```

Of course the comment should be kept somewhere. React Concurrent is still not in stable, but will probably come out in the next major release.